Bottom: 4e6c2f76c5e95e662de0f2c9c8404295fd64591f
Top:    8acc4f5f1cd0cc3ae2ffea5ff8583c029904c895
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 15:55:55 +0000

polygon1.c: Fix unitialised output polygon upon when hitting allocation failure

This was a bug present in the poly_Boolean() routine. We need to set the output
polygon to NULL if we are going to return early on memory failure. This was
noticed during general review after observing problems due to our copy routines
incorrectly returning FALSE when copying a NULL polygon.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 33a2212..511bd21 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2326,6 +2326,8 @@ poly_Boolean (const POLYAREA * a_org, const POLYAREA * b_org,
 {
   POLYAREA *a = NULL, *b = NULL;
 
+  *res = NULL;
+
   if (!poly_M_Copy0 (&a, a_org) || !poly_M_Copy0 (&b, b_org))
     return err_no_memory;
